### DISCUSSION TITLE
[Mailer] Send email with DSN from database?

### DISCUSSION THREAD

**Discussion by sofigio** [symfony/symfony#44763] [Q&A]
I'm reading [the Sending Emails with Mailer](https://symfony.com/doc/current/mailer.html) tutorial. 
I see many transports have been implemented, all of them must have their settings hardcoded into the .env file.

Nobody thought about a case where the admin will choose the SMTP setting of their application from its control panel, thus saving the mailer host, port, username and password to the database? 

I have already a Settings class with `get` and `set` method to read and write the application settings to db, how can I inject this class to the mailer to read the settings?

**Comment by norkunas** [comment]
You could use `Symfony\Component\Mailer\Transport (mailer.transport_factory)` to create your mailer on runtime based on your settings

**Comment by irayoun** [comment]
As @norkunas says, you can @sofigio use _Transport_ class to set up your SMTP parameters from DB like:

`
$dsn = "smtp://" . $username . ":" . $password . "@" . $host. ":" . $port;
$transport = Transport::fromDsn($dsn);
$customMailer = new Mailer($transport);
`

I hope this will be useful to you :)
